src/hotspot/os/aix/porting_aix.cpp

```diff
@@ -1049,14 +1050,25 @@ void* Aix_dlopen(const char* filename, int Flags, const char** error_report) {
     return nullptr;
   }
   else {
+    // extract member string if exist duplicate it and store pointer of it
+    // if member does not exist store nullptr
+    char* member = nullptr;
+    const char* substr;
+    if (filename[strlen(filename) - 1] == ')' && (substr = strrchr(filename, '('))) {
+      member = os::strdup(substr);
+    }
+
     unsigned i = 0;
     TableLocker lock;
     // check if library belonging to filename is already loaded.
     // If yes use stored handle from previous ::dlopen() and increase refcount
     for (i = 0; i < g_handletable_used; i++) {
       if ((p_handletable + i)->handle &&
           (p_handletable + i)->inode == libstat.st_ino &&
-          (p_handletable + i)->devid == libstat.st_dev) {
+          (p_handletable + i)->devid == libstat.st_dev &&
+          (((p_handletable + i)->member == member) ||
```

Comment:
If this ever fires (wildly unlikely), its a bug. You are comparing a stack address with something that lives in C heap right now.

Reply:
I discussed this with Martin. The meaning of the line is: Check if both are nullptr.
My original implementation was:
(((p_handletable + i)->member == nullptr && member == nullptr) ||
Martin favoured the shorter version above which will have the same result.

Reply:
It is, to me, not obvious where the member is reset. We delete it, but don't set the pointer to NULL.So, its not obvious that entry->member could not point to a former stale c heap address.

In any case, I prefer the cleanly written out logic. The generated code will be the same anyway, so this brevity just obfuscates your intent and confuses casual readers.

Reply:
Back to the roots.

Reply:
I think that Thomas' point is that `member = nullptr;` is missing after the `free` operation. I agree with that.